### PR TITLE
Magneto hard fork

### LINKED ISSUE #2314: Magneto upgrade
### Description
Perform the steps necessary for the Magneto upgrade.

See https://github.com/ethereumclassic/ECIPs/issues/424 for context.

### PR DESCRIPTION
**Pull request by atoulme**
Signed-off-by: Antoine Toulme <antoine@lunar-ocean.com>

## PR description
Adds the blocks and the changes for the Magneto hard fork.

## Fixed Issue(s)
Fixes #2314 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by shemnon — CHANGES_REQUESTED** [review]
Relevant fork IDs in ForkIdTest.java need to be updated.

File: config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java [hunk 1]
```diff
@@ -317,6 +317,11 @@ public OptionalLong getThanosBlockNumber() {
     return getOptionalLong("thanosblock");
   }
 
+  @Override
+  public OptionalLong getMagnetoBlockNumber() {
+    return getOptionalLong("magnetoBlock");
```
Comment (shemnon) [inline config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java:322]: Should be lowercase.

**Comment by shemnon** [comment]
Also update ProtocolScheduleBuilder#validateClassicForkOrdering, looks to be missing Thanos as well there.

**Comment by atoulme** [comment]
Thanks Danno, addressed your comments now.

**Comment by shemnon** [comment]
This test _should_ keep it from occurring again if we follow the pattern of ^F old fork block->do what was there.

https://github.com/hyperledger/besu/pull/2317

**Comment by atoulme** [comment]
Let's get #2317 in first.

**Review by shemnon — APPROVED** [review]

**Comment by atoulme** [comment]
Merged #2317 into this PR.

**Pull request merged**